android/library/src/main/java/com/liferay/mobile/screens/context/SessionContext.java

```diff
@@ -125,6 +133,40 @@ public static void loadStoredCredentials(CredentialsStorageBuilder.StorageType s
 		}
 	}
 
+	public static void relogin(LoginListener loginListener) {
+		if (_currentUserSession == null || _currentUserSession.getAuthentication() == null) {
+			loginListener.onLoginFailure(new AccessControlException("Missing user attributes"));
+		}
+
+		refreshUserAttributes(loginListener, new SessionImpl(LiferayServerContext.getServer(),
+			_currentUserSession.getAuthentication()));
+	}
+
+	private static void refreshUserAttributes(final LoginListener loginListener, final Session session) {
+		Executor.execute(new Runnable() {
+			@Override
+			public void run() {
+				try {
+					CurrentUserConnector currentUserConnector = ServiceProvider.getInstance().getCurrentUserConnector(session);
+					JSONObject jsonObject = currentUserConnector.getCurrentUser();
+					if (jsonObject != null) {
+
+						User user = new User(jsonObject);
+						_currentUser = user;
```

Comment:
@nhpatt if the user was read from the store, that stored data is not updated with the new user retrieved.